src/graph/savegraphaction.pas

```diff
@@ -150,7 +202,44 @@ constructor TCustomSaveGraphAction.Create(AOwner: TComponent);
   FGraphSize := TSize.Create(1024, 768);
   OnExecute := @SaveGraphExecute;
   Caption := 'Save as ...';
+  FCount := 0;
+end;
+
+destructor TCustomSaveGraphAction.Destroy;
+var
+  i: integer;
+begin
+  // remove charts before freeing
+  for i := 0 to high(FCharts) do
+    RemoveComponent(FCharts[i]);
+  inherited Destroy;
+end;
+
+// create a save file name from base name and optional text, which should be stratum value (not label)
+function GetSaveChartFilename(AFileName: UTF8String; AValue: UTF8String): UTF8String;
```

Comment:
I would really like that we used to FPC methods rather than our own implementation here.

Reply:
This helper function now simply inserts whatever value is saved in the title configuration without checking it.